### PR TITLE
Sales report user column link now accounts for guest user purchases

### PR DESCRIPTION
**Pull request by DevinWalker**
Fix for #3611

### CODE REVIEW

**Comment by DevinWalker** [comment]
@cklosowski @pippinsplugins I also fixed two broken links in c2337fa5fb02d06e60a7d92e4af22fc8aabf6f59

**Comment by DevinWalker** [comment]
Oops accidentally closed

File: includes/admin/reporting/class-sales-logs-list-table.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function column_default( $item, $column_name ) {
 
 			case 'user_id' :
 				return '<a href="' .
-					admin_url( 'edit.php?post_type=download&page=edd-payment-history&user=' . urlencode( $item['user_id'] ) ) .
+				       admin_url( 'edit.php?post_type=download&page=edd-payment-history&user=' . ( ! empty( $item['user_id'] ) ? urlencode( $item['user_id'] ) : edd_get_payment_user_email( $item['payment_id'] ) ) ) .
```
Comment (pippinsplugins) [inline includes/admin/reporting/class-sales-logs-list-table.php]: Could you separate this out into two lines? having ternary logic inline gets difficult to read.

Comment (DevinWalker) [inline includes/admin/reporting/class-sales-logs-list-table.php]: Yeah

Comment (DevinWalker) [inline includes/admin/reporting/class-sales-logs-list-table.php]: @pippinsplugins how do you prefer I format the admin_url param? if/else statement above return?

Comment (pippinsplugins) [inline includes/admin/reporting/class-sales-logs-list-table.php]: No need for an if/else, just setup the `user` parameter with a ternary (like you have) on the line above.

**Comment by DevinWalker** [comment]
@pippinsplugins I've maked it clearer

**Comment by pippinsplugins** [comment]
Much better :)

**Pull request merged**